pkg/manifestgen/manifestgen_test.go

```diff
@@ -136,7 +136,7 @@ func TestManifestGeneratorWithOstreeCommit(t *testing.T) {
 	assert.Contains(t, osbuildManifest.String(), expectedSha256)
 }
 
-func fakeDepsolve(cacheDir string, packageSets map[string][]rpmmd.PackageSet, d distro.Distro, arch string) (map[string]dnfjson.DepsolveResult, error) {
+func fakeDepsolve(cacheDir string, warningsOutput io.Writer, packageSets map[string][]rpmmd.PackageSet, d distro.Distro, arch string) (map[string]dnfjson.DepsolveResult, error) {
```

Comment:
Let's add a test for the new depsolveOuput too:
1. AIUI warningsOutput is depsolveOutput, lets tweak the name to avoid confusion with the existing warningsOutputs which contolls something else
2. in the fakeDepsolve, lets write something to the depsolveOuptut (if non-nil)
3. and a test that passes a buffer where we check that the output ends up. 
4. it does not test much (but it also shouldn't be much work) but its still useful to ensure the plumbing is correct, i.e. l115,l169

Reply:
Yes, let's add a test. I must've forgotten the name because initially I was passing along `warningsOutput` instead of having a separate one (which was bad).

Reply:
I've added the test (I think mostly as you described) :)